collectors/o365/o365_collector.js

```diff
@@ -168,29 +169,17 @@ class O365Collector extends PawsCollector {
 
             return callback(null, logs, newState, newState.poll_interval_sec);
         }).catch(err => {
-            if (err.message) {
-                let message = err.message;
-                // set errorCode if not available in error object to showcase client error on DDMetric
-                try {
-                    let error = JSON.parse(message.slice(message.indexOf('{'), message.lastIndexOf('}') + 1));
-                    err.errorCode = error.error ? error.error : error.error_codes[0];
-                    if (error.error_codes) {
-                        if (error.error_codes[0] === 7000215) {
-                            return callback("Error code [7000215]. Invalid client secret is provided.");
-                        }
-                        if (error.error_codes[0] === 700016) {
-                            return callback("Error code [700016]. Invalid client ID is provided.");
-                        }
-                        if (error.error_codes[0] === 90002) {
-                            return callback("Error code [90002]. Please make sure you have the correct tenant ID or this may happen if there are no active subscriptions for the tenant. Check with your subscription administrator.");
-                        }
-                    } 
-                } catch (exception) {
-                    return callback(err);
-                }
+            // set errorCode to showcase client error on DDMetric
+            if (typeof err === 'object') {
+                err.errorCode = err.code ? err.code : (err.statusCode ? err.statusCode : err.status);
             }
-            console.error(`O365000003 Error in collection: ${err.message}`);
-            return callback(err);
+            let newState = this._handleMSManagementApiError(err, state);
+            if (newState) {
+                return callback(null, [], newState, newState.poll_interval_sec);
+            }
+            
+            this._setUserUnderstandableErrorMessage(err, callback);
```

Comment:
this function is not only formats messages. ANd I'd prefer to keep final callback in here.  Let's replace/rename it with
```
if (!err.code && err.message) 
    const msg = this._formatErrorMessage(err);
    console.error(`O365000003 Error in collection: ${err.message}`);
    return callback(msg)
else 
 return callback(null)
```

Reply:
sure, I will rename the function name to _formatErrorMessage.